### PR TITLE
Improve API ergonomics, updated examples and readme

### PR DESCRIPTION
**Pull request by davidzhao**

### CODE REVIEW

**Review by keepingitneil — APPROVED** [review]

**Review by theomonnom — APPROVED** [review]
ty

File: livekit-api/livekit/api/access_token.py [hunk 1]
```diff
@@ -69,10 +70,16 @@ class Claims:
...
+    def __init__(
+        self,
+        api_key: str = os.getenv("LIVEKIT_API_KEY"),
+        api_secret: str = os.getenv("LIVEKIT_API_SECRET"),
+    ) -> None:
         self.api_key = api_key  # iss
         self.api_secret = api_secret
         self.claims = Claims()
+        if api_key is None or api_secret is None:
+            raise ValueError("api_key and api_secret must be set")
```
Comment (theomonnom) [inline livekit-api/livekit/api/access_token.py:82]:
```suggestion
        api_key: str = os.getenv("LIVEKIT_API_KEY", ""),
        api_secret: str = os.getenv("LIVEKIT_API_SECRET", ""),
    ) -> None:
        self.api_key = api_key  # iss
        self.api_secret = api_secret
        self.claims = Claims()
        if not api_key or not api_secret :
            raise ValueError("api_key and api_secret must be set")
```

**Review by theomonnom — CHANGES_REQUESTED** [review]

File: livekit-api/livekit/api/access_token.py (hunk 1 above)
Comment (theomonnom) [reply livekit-api/livekit/api/access_token.py:82]: Keep typing OK (str)

File: livekit-api/livekit/api/room_service.py [hunk 2]
```diff
@@ -7,8 +7,8 @@
 
 
 class RoomService(Service):
-    def __init__(self, host: str, api_key: str, api_secret: str):
-        super().__init__(host, api_key, api_secret)
+    def __init__(self, url: str, api_key: str, api_secret: str):
```
Comment (theomonnom) [inline livekit-api/livekit/api/room_service.py:10]: Should we also get env here?

**Review by theomonnom — APPROVED** [review]

**Pull request merged**